### PR TITLE
Reduce warning and info messages in node log files (Artemis)

### PR DESCRIPTION
**Pull request by josecoll**
Provide useful information when Notary detects conflicting states.

### CODE REVIEW

File: core/src/main/kotlin/net/corda/flows/NotaryFlow.kt [hunk 1]
```diff
@@ -134,6 +134,7 @@ object NotaryFlow {
                 }
                 if (conflicts.isNotEmpty()) {
                     // TODO: Create a new UniquenessException that only contains the conflicts filtered above.
+                    logger.warn("Notary conflicts for $txId: $conflicts")
```
Comment (shamsasari) [inline core/src/main/kotlin/net/corda/flows/NotaryFlow.kt]: Is the exception not logged on the other side? I remember you had the problem where the exception message wasn't being printed, or similar.

File: config/dev/log4j2.xml [hunk 2]
```diff
@@ -57,5 +57,8 @@
             <AppenderRef ref="Console-Appender-Println"/>
             <AppenderRef ref="RollingFile-Appender" />
         </Logger>
+        <Logger name="org.apache.activemq.artemis" level="error" additivity="false">
```
Comment (shamsasari) [inline config/dev/log4j2.xml]: not `warning`?

File: node/src/main/kotlin/net/corda/node/services/messaging/ArtemisMessagingServer.kt [hunk 3]
```diff
@@ -576,7 +577,7 @@ class NodeLoginModule : LoginModule {
         val password = String(passwordCallback.password ?: throw FailedLoginException("Password not provided"))
         val certificates = certificateCallback.certificates
 
-        log.info("Processing login for $username")
+        log.trace { "Processing login for $username" }
```
Comment (shamsasari) [inline node/src/main/kotlin/net/corda/node/services/messaging/ArtemisMessagingServer.kt]: debug?

File: config/dev/log4j2.xml (hunk 2 above)
Comment (josecoll) [inline config/dev/log4j2.xml]: The problem with `warning` is you get hundreds of these:
```

[WARN ] 2017-03-28T13:32:27,861 [Thread-1 (activemq-netty-threads-1526894424)] core.server.checkDeadLetterAddressAndExpiryAddress - AMQ222165: No Dead Letter Address configured for queue clients.user1.rpc.9071990153569920168 in AddressSettings
[WARN ] 2017-03-28T13:32:27,861 [Thread-1 (activemq-netty-threads-1526894424)] core.server.checkDeadLetterAddressAndExpiryAddress - AMQ222166: No Expiry Address configured for queue clients.user1.rpc.9071990153569920168 in AddressSettings
```

File: core/src/main/kotlin/net/corda/flows/NotaryFlow.kt (hunk 1 above)
Comment (josecoll) [inline core/src/main/kotlin/net/corda/flows/NotaryFlow.kt]: Quite possible although I was only seeing a 'null' value on the client side.
However, from an operational support perspective, it is much easier to grep the Notary Logs for such errors (as a 1st step indicator of an issue)

File: node/src/main/kotlin/net/corda/node/services/messaging/ArtemisMessagingServer.kt (hunk 3 above)
Comment (josecoll) [inline node/src/main/kotlin/net/corda/node/services/messaging/ArtemisMessagingServer.kt]: yep - I can change that.
For some reason, I always consider `trace` a level higher than `debug` but it isn't according to the log4j docs ;-)

Quote:
The DEBUG Level designates fine-grained informational events that are most useful to debug an application.
The TRACE Level designates finer-grained informational events than the DEBUG

File: config/dev/log4j2.xml (hunk 2 above)
Comment (shamsasari) [inline config/dev/log4j2.xml]: I think we should fix these issues rather than hiding them.

Comment (josecoll) [inline config/dev/log4j2.xml]: Its not really something we need to address as these are temporary (transient) queues - which don't really need a DLQ

Comment (mikehearn) [inline config/dev/log4j2.xml]: Yeah, the fix for this issue is to submit a patch to Artemis that tweaks the logging level to be more appropriate (or file an issue), and we can suppress Artemis warnings from core.server. checkDeadLetterAddressAndExpiryAddress in the meantime perhaps?

Comment (josecoll) [inline config/dev/log4j2.xml]: Yep - thats a good compromise.
Let me do that.

File: core/src/main/kotlin/net/corda/flows/NotaryFlow.kt (hunk 1 above)
Comment (josecoll) [inline core/src/main/kotlin/net/corda/flows/NotaryFlow.kt]: Perhaps its worth revisiting the FlowException propagation on Notary Exception - to make sure values are propagated correctly.
Do we have a simulated Double Spend JUnit to reproduce this?

Comment (shamsasari) [inline core/src/main/kotlin/net/corda/flows/NotaryFlow.kt]: `NotaryServiceTests` tests double spends by checking `NotaryException` is thrown by the start flow future.

**Pull request merged**